### PR TITLE
Swap v2  Code Refactoring, Mapped code according to Uniswap V2 and addedTest-Cases 

### PR DESCRIPTION
**Pull request by ashi20192020**

### CODE REVIEW

File: frame/social-swap2/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,10 @@ sp-core = { default-features = false, version = '3.0.0', path = '../../primitive
 pallet-timestamp = { default-features = false, version = '3.0.0', path = '../timestamp' }
 pallet-assets = { default-features = false, version = '3.0.0', path = '../assets' }
 
+[dev-dependencies]
+serde = "1.0.101"
```
Comment (andor0) [inline frame/social-swap2/Cargo.toml]: we using `serde 1.0.124` in the project https://github.com/social-network/blockchain/blob/master/Cargo.lock#L8211-L8218

File: frame/social-swap2/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+			pallet_assets::Module::<Test>::total_supply(ASSET_ID),
+			0
+		);
+		assert_eq!(
+			SocialSwap2::mint(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT2
+			),
+			Ok(())
+		);
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs:40]: can use `assert_ok!` macro here
```
assert_ok!(
			SocialSwap2::mint(
				Origin::signed(ACCOUNT1),
				ACCOUNT2
			)
		);
```

File: frame/social-swap2/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+#[test]
+fn test_burn_should_not_work() {
+	new_test_ext().execute_with(|| {
+
+		assert_ne!(SocialSwap2::mint(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT3
+			),Ok(()));
+
+	});
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs:83]: can use `assert_noop!` to check error here
```
assert_noop!(SocialSwap2::mint(
				Origin::signed(ACCOUNT1),
				ACCOUNT3
			),
                        Error::<Test>::SomeError
                 );
```

File: frame/social-swap2/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+		);
+
+
+		assert_eq!(
+			SocialSwap2::burn(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT2
+			),
+			Ok(())
+		);
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs:115]: can use `assert_ok!` macro here

File: frame/social-swap2/src/tests.rs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+	SocialSwap2::initialize(Origin::root(), FEE_TO, ADDRESS0, TREASURY, TOKEN0, TOKEN1);
+	pallet_assets::Module::<Test>::transfer(&ASSET_ID, &ACCOUNT1, &TOKEN0, token_0_amount);
+	pallet_assets::Module::<Test>::transfer(&ASSET_ID, &ACCOUNT1, &TOKEN1, token_1_amount);
+	assert_eq!(
+		SocialSwap2::mint(
+			Origin::signed(ACCOUNT1),
+			ACCOUNT2
+		),
+		Ok(())
+	);
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs:256]: can use `assert_ok!` macro here

File: frame/social-swap2/src/tests.rs [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+		assert_eq!(SocialSwap2::swap(
+			Origin::signed(ACCOUNT1),
+			0,
+			expected_output_amount,
+			ACCOUNT2,
+			"0x".encode()
+		),
+				   Ok(())
+		);
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs:216]: can use `assert_ok!` macro here

File: frame/social-swap2/src/tests.rs [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+#[test]
+fn test_swap_should_work() {
+	new_test_ext().execute_with(|| {
+		let token_0_amount: u128  = 5_000_000_000_000_000_000;
+		let token_1_amount: u128  = 10_000_000_000_000_000_000;
+		add_liquidity(token_0_amount, token_1_amount);
+		let expected_output_amount: u128 = 1662497915624478906u128;
+		let sawp_amount: u128  = 1_000_000_000_000_000_000;
```
Comment (andor0) [inline frame/social-swap2/src/tests.rs]: is `sawp_amount` typo?
i.e. `sawp_amount` =>  `swap_amount` ?

**Review by andor0 — CHANGES_REQUESTED** [review]

File: frame/social-swap2/src/tests.rs (hunk 3 above)
Comment (ashi20192020) [reply frame/social-swap2/src/tests.rs:83]: @andor0 

Ideally it should be 

```
assert_noop!(
			SocialSwap2::mint(
				Origin::signed(ACCOUNT1),
				ACCOUNT3
			),
			Error::<Test>::InsufficientLiquidityMinted
		);
```

But its giving weird error.
		
```
		---- tests::test_burn_should_not_work stdout ----
thread 'tests::test_burn_should_not_work' panicked at 'assertion failed: `(left == right)`
  left: `[128, 219, 156, 72, 65, 174, 164, 236, 63, 108, 115, 17, 130, 98, 41, 26, 225, 203, 75, 115, 97, 222, 192, 210, 45, 21, 176, 145, 212, 197, 126, 200]`,
 right: `[239, 236, 170, 140, 85, 84, 36, 71, 249, 25, 167, 223, 40, 139, 34, 146, 239, 90, 7, 71, 111, 192, 3, 123, 43, 134, 173, 115, 110, 201, 173, 166]`', frame/social-swap2/src/tests.rs:78:5
```

File: frame/social-swap2/src/lib.rs [hunk 8]
```diff
@@ -184,11 +189,12 @@ decl_module! {
...
 
             if amount0_out > 0u32.into() {
-                Self::safe_transfer(&token0, &to, amount0_out)?
+				T::FungibleToken::transfer(&social_token_id, &token0, &to, amount0_out);
             }
             if amount1_out > 0u32.into() {
-                Self::safe_transfer(&token1, &to, amount1_out)?
+				T::FungibleToken::transfer(&social_token_id, &token1, &to, amount1_out);
             }
+			//assert_eq!(true, false);
```
Comment (andor0) [inline frame/social-swap2/src/lib.rs]: forgot to delete it?

**Review by andor0 — APPROVED** [review]

**Pull request merged**